### PR TITLE
Add Email claim

### LINKED ISSUE #18348: User.FindFirstValue(ClaimTypes.Email) returns always null.
I am trying to get the logged in user's email from the claim using `User.FindFirstValue(ClaimTypes.Email)` but it always returning `null`. Look at the followings:

     string email = User.FindFirstValue(ClaimTypes.Email); // Always null. Not returning logged in user's Email
     string userName = User.FindFirstValue(ClaimTypes.Name); // Returning logged in user's UserName
     string userId = User.FindFirstValue(ClaimTypes.NameIdentifier); // // Returning logged in user's Id

Why `User.FindFirstValue(ClaimTypes.Email)` returning null instead of logged in user's email?

**Note: `User.FindFirst(ClaimTypes.Email)` also returning `null` while `User.FindFirst(ClaimTypes.Name)` and `User.FindFirst(ClaimTypes.NameIdentifier)`; returning expected values.**

### PR DESCRIPTION
**Pull request by Kahbazi**
Addresses #18348

### CODE REVIEW

File: src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs [hunk 1]
```diff
@@ -76,11 +76,17 @@ protected virtual async Task<ClaimsIdentity> GenerateClaimsAsync(TUser user)
         {
             var userId = await UserManager.GetUserIdAsync(user);
             var userName = await UserManager.GetUserNameAsync(user);
+            var email = await UserManager.GetEmailAsync(user);
```
Comment (HaoK) [inline src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs]: You probably have to check whether email is supported before doing this, similar to the claims stamp check below

Comment (Kahbazi) [reply src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs]: @Haok Done! Do I need to check weather email claim is null or empty before adding it as a claim?

Comment (HaoK) [reply src/Identity/Extensions.Core/src/UserClaimsPrincipalFactory.cs]: It doesn't hurt so why not

**Comment by Kahbazi** [comment]
@Hoak Is the build process stuck?

**Review by HaoK — APPROVED** [review]

**Comment by HaoK** [comment]
Thanks @Kahbazi

**Comment by stephenwilson1984** [comment]
Hi guys,

I'm trying to use this in .NET 6 using VS 2022 Preview and it still seems to be broken as per the original bug.

`principal.FindFirstValue(ClaimTypes.NameIdentifier);` - Works fine.
`principal.FindFirstValue(ClaimTypes.Email);` - Returns null.

I can see from the list of claims that only `NameIdentifier` seems to be added.

Claims List

**Comment by m3c-ode** [comment]
If I define my claims as such:
```
List<Claim> claims = new()
            {
                new Claim("name", userModel.UserName),
                new Claim("email", userModel.Email),
                new Claim("id", userModel.Id),
            };
```

`var userEmailClaim = User.FindFirst("email").Value!` iis always null and empty.  I don't understand what the fix is?

**Pull request merged**